### PR TITLE
Extract Hadoop path conversion to ContentKey into util class

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.Arrays;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.Path;
+import org.projectnessie.model.ContentKey;
+
+/**
+ * Utility class that produces a {@link ContentKey} instance from a Hadoop {@link Path} or a file
+ * path string.
+ */
+public class ContentKeyUtil {
```
Comment (nastra) [inline clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java:28]: given that this is really only required for Delta, I decided to put this into the deltalake package

File: clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  public static ContentKey fromFilePathString(String path) {
+    Preconditions.checkArgument(null != path && !path.isEmpty(), INVALID_PATH_MSG, path);
+    // we can't simply do a path.split("/") as that would result with
+    // "/tmp/a/b/c in a string array where the first element is an empty string,
+    // which isn't supported by ContentKey
+    String[] elements = StringUtils.stripEnd(StringUtils.stripStart(path, "/"), "/").split("/");
+    Preconditions.checkState(
+        elements.length > 0 && !Arrays.asList(elements).contains(""), INVALID_PATH_MSG, path);
```
Comment (XN137) [inline clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java:46]: nit: isnt the 2nd condition already checked inside `ContentKey` ?

File: clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.Arrays;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.Path;
+import org.projectnessie.model.ContentKey;
+
+/**
+ * Utility class that produces a {@link ContentKey} instance from a Hadoop {@link Path} or a file
+ * path string.
+ */
+class ContentKeyUtil {
```
Comment (XN137) [inline clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java:28]: not sure if this is a good idea, but in order to not pollute the namespace, should this be `DeltaContentKeyUtil` maybe?
or `DeltaContentKey`.

nit: maybe add empty private constructor (for utility classes)

**Review by XN137 — DISMISSED** [review]
thanks for doing this followup, lgtm

File: clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java (hunk 2 above)
Comment (nastra) [reply clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java:46]: it is, but the error msg can be a bit misleading after the path string was modified and passed to ContentKey

File: clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java (hunk 3 above)
Comment (nastra) [reply clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java:28]: done

**Review by XN137 — DISMISSED** [review]

**Review by XN137 — APPROVED** [review]

**Pull request merged**